universe/com.goal_smart/manifest.tt

```diff
@@ -14,42 +45,107 @@ class @com.goal_smart
   entity soccer_league #_[description="Soccer Leagues in the World"];
   entity soccer_player #_[description="Soccer Players in the World"];
   
-  monitorable list query standings(in req league_id: Entity(com.goal_smart:soccer_league) #_[prompt="what league standings do you want out check out?"],
-                        out team : String,
+  monitorable list query standings(in req league_id: Entity(com.goal_smart:soccer_league) #_[prompt="what league standings do you want out check out"],
+                        out team : Entity(com.goal_smart:soccer_team),
                         out rank: Number,
-                        out form: String,
+                        out form: Array(String),
                         out points: Number)
   #_[canonical="league standings, league rankings, soccer rankings, league table, soccer table"]
   #_[confirmation="league standings of $league"]
   #_[result = ["${rank} . ${team} Form: ${form} Points: ${points}"]]
   #[poll_interval=3600000ms]
   #[doc="Get the standings of a league"];
 
-  monitorable list query teamFixtures(in req team_id: Entity(com.goal_smart:soccer_team) #_[prompt="what team do you want out check out?"],
-                        out league : String,
-                        out team1: String,
-                        out team2: String,
+
+
+
+
+
+
+
+
+  monitorable query firstPlace(in req league_id: Entity(com.goal_smart:soccer_league) #_[prompt="what league do you want to see this info for"],
+                        out team : Entity(com.goal_smart:soccer_team),
+                        out points: Number)
+  #_[canonical="league winners, league leaders, league firsts"]
+  #_[confirmation="first place of $league"]
+  #_[result = ["${team} is in first place with ${points} points"]]
+  #[poll_interval=3600000ms]
+  #[doc="Get the first place team of a league"];
```

Comment:
You don't need these functions, and in fact you don't want them.
You can get the same effect using sort and index operators on the standings:
```
sort(points desc of @com.goal_smart.standings(...))[1]
```
or in fact using the rank property:
```
@com.goal_smart.standings(), rank == 1
```

(in fact, the fact that you have two ways to express the same concept is a bad idea because it will cause confusion in the model, you should probably make the `rank` parameter not filterable)

If you haven't already, I highly recommend reading the "database operations" part of the [ThingTalk guide](https://wiki.almond.stanford.edu/thingtalk/guide), because what you're writing is essentially a lot of database queries.